Raven.Database/Raft/ClusterManager.cs

```diff
@@ -234,6 +290,34 @@ public void Dispose()
 
             aggregator.ThrowIfNeeded();
         }
+
+        private bool disableReplicationStateChecks;
+        private int disableReplicationStateChecksLocker;
+        public void HandleClusterConfigurationChanged(ClusterConfiguration configuration)
+        {
+            if (Interlocked.CompareExchange(ref disableReplicationStateChecksLocker, 1, 0) == 1)
+                return;
+            try
+            {
+                //same value nothing to do
+                if (configuration.DisableReplicationStateChecks == disableReplicationStateChecks)
+                    return;
+                if (configuration.DisableReplicationStateChecks)
+                {
+                    Engine.StateChanged -= OnRaftEngineStateChanged;
+                    SaflyDisposeOfTimer();
+                }
+                else
+                {
+                    Engine.StateChanged += OnRaftEngineStateChanged;
```

Comment:
You are doing that all the time, and if you are the leader and the state changed to false and then true, it looks like the timer is disposed and that is it.